### ISSUE TITLE
Investigate the usage of `tracing_subscriber`

### ISSUE THREAD

**Issue by ClementWalter** [kkrt-labs/kakarot-rpc#177]
The rpc `main.rs` file defined

```
use dotenv::dotenv;
use eyre::{eyre, Result};
use kakarot_rpc::run_server;
use kakarot_rpc_core::client::KakarotClientImpl;
use starknet::core::types::FieldElement;
use tracing_subscriber::util::SubscriberInitExt;

#[tokio::main]
async fn main() -> Result<()> {
    dotenv().ok();

    let filter = tracing_subscriber::EnvFilter::try_from_default_env()?
        .add_directive("jsonrpsee[method_call{name = \"eth_chainId\"}]=trace".parse()?);
    tracing_subscriber::FmtSubscriber::builder().with_env_filter(filter).finish().try_init()?;
```

but we actually didn't use this elsewhere nor were able to justify this.

So it has been removed and this issue aims for recording the fact that we'd need to investigate why this is added in the boilerplate, and how to use.

**Comment by bajpai244** [comment]
@ClementWalter I can work on this investigation.

**Comment by Eikix** [comment]
What is the status of this issue? Does this need to be closed, as we're discussing tracing in a dedicated discussion.

Also, are we properly addressing what tracing is?

Tracing API -> querying execution traces of Kakarot from Starknet RPC, and showing them in Starknet trace form, or EVM form.

Tracing in jsonrpsee -> logging?

**Comment by bajpai244** [comment]
@Eikix we can close this.

- The issue was created to understand the code snippet [here](https://github.com/kkrt-labs/kakarot-rpc/blob/185d8285790cd5bd3610f29f6bbaccfb22f64560/crates/eth-rpc/src/main.rs#L14C1-L14C1) was doing, as nowhere else in the codebase we could find anything related to it.
- Then we figured out that since RPC runs in an async environment, hence to get meaningful logs we need tracing. So, the original quest from what I remember is done, which was that we didn't know why this snipped existed.
- This issue was for tracing _subsriber, which is related `jsonrpsee`.
- I have created a separate discussion for Tracing API [here](https://github.com/kkrt-labs/kakarot-rpc/discussions/196).

TLDR; 

We now know that `tracing_subsriber` has a real utility and is not some junk code living in the repo, so I guess we can close it.

**Comment by Eikix** [comment]
> @Eikix we can close this.
> 
> - The issue was created to understand the code snippet [here](https://github.com/kkrt-labs/kakarot-rpc/blob/185d8285790cd5bd3610f29f6bbaccfb22f64560/crates/eth-rpc/src/main.rs#L14C1-L14C1) was doing, as nowhere else in the codebase we could find anything related to it.
> - Then we figured out that since RPC runs in an async environment, hence to get meaningful logs we need tracing. So, the original quest from what I remember is done, which was that we didn't know why this snipped existed.
> - This issue was for tracing _subsriber, which is related to `jsonrpsee`.
> - I have created a separate discussion for Tracing API [here](https://github.com/kkrt-labs/kakarot-rpc/discussions/196).
> 
> TLDR; 
> 
> We now know that `tracing_subsriber` has a real utility and is not some junk code living in the repo, so I guess we can close it.

But does that mean we don't use jsonrpsee tracing subscriber anymore?

**Comment by bajpai244** [comment]
> > @Eikix we can close this.
> > 
> > * The issue was created to understand the code snippet [here](https://github.com/kkrt-labs/kakarot-rpc/blob/185d8285790cd5bd3610f29f6bbaccfb22f64560/crates/eth-rpc/src/main.rs#L14C1-L14C1) was doing, as nowhere else in the codebase we could find anything related to it.
> > * Then we figured out that since RPC runs in an async environment, hence to get meaningful logs we need tracing. So, the original quest from what I remember is done, which was that we didn't know why this snipped existed.
> > * This issue was for tracing _subsriber, which is related to `jsonrpsee`.
> > * I have created a separate discussion for Tracing API [here](https://github.com/kkrt-labs/kakarot-rpc/discussions/196).
> > 
> > TLDR;
> > We now know that `tracing_subsriber` has a real utility and is not some junk code living in the repo, so I guess we can close it.
> 
> But does that mean we don't use jsonrpsee tracing subscriber anymore?

- We do use the tracing subscriber, the logs that we see when the RPC starts are possible because of it.
- This issue was to investigate, what is the utility of the tracing subscriber, now we know the utility of the subscriber and we know that we shouldn't remove it, as it will disable the logging we have as of now.

**Comment by Eikix** [comment]
```rust
let filter = tracing_subscriber::EnvFilter::try_from_default_env()?
        .add_directive("jsonrpsee[method_call{name = \"eth_chainId\"}]=trace".parse()?);
    tracing_subscriber::FmtSubscriber::builder().with_env_filter(filter).finish().try_init()?;
```

Did we answer the question: "Why does it look like the tracing is only done on eth_chainId"?

**Comment by bajpai244** [comment]
@Eikix we have removed this the directive from "eth_chainId".
